### ISSUE TITLE
Template variable expansion uses free-identifier=? instead of bound-identifier=?

### ISSUE THREAD

**Issue by ktakashi** [ktakashi/sagittarius-scheme#159]
**[Original report](https://bitbucket.org/ktakashi/sagittarius-scheme/issues/154) by Takashi Kato (Bitbucket: [ktakashi](https://bitbucket.org/ktakashi), GitHub: [ktakashi](https://github.com/ktakashi)).**

----------------------------------------

The following code returns incorrect values:

```
#!scheme
(import (rnrs))

(define-syntax bar
  (lambda (x)
    (syntax-case x ()
      ((_ (ts ...) (b ...))
       #'(let-syntax ((buz (lambda (x)
                             (syntax-case x ()
                               ((_ ts ...) #'(list ts ...))))))
           (buz b ...)))
      ((_ (t* ...) (b ...) a as ...)
       #'(bar (t* ... t) (b ...) as ...)))))

(bar () (1 2 3) 1 2 3)
;; -> (3 3 3)
;; must be (1 2 3)
```
This is because `free-identifier=?` returns `#t` with all temporary identifiers `t`.

**Comment by ktakashi** [comment]
**Original comment by Takashi Kato (Bitbucket: [ktakashi](https://bitbucket.org/ktakashi), GitHub: [ktakashi](https://github.com/ktakashi)).**

----------------------------------------

Preserving pattern variable during syntax-case rewrite (Fixes #154)

**Comment by ktakashi** [comment]
**Original comment by Takashi Kato (Bitbucket: [ktakashi](https://bitbucket.org/ktakashi), GitHub: [ktakashi](https://github.com/ktakashi)).**

----------------------------------------

(Was: free-identifier=? returns #t even though it not the same bindings)
The fix was actually not related to `free-identifier=?` but incorrect usage of `free-identifier=?`
